### PR TITLE
Use GitHub Actions for CI build

### PR DESCRIPTION
**Pull request by orien**
#### Context

With wait times getting longer on our previous CI provider, we have decided to use GitHub Actions for our open-source projects.

#### Change

- Define a 'Tests' action in [`.github/workflows/test.yml`](https://github.com/envato/event_sourcery/pull/228/files#diff-faff1af3d8ff408964a57b2e475f69a6b7c7b71c9978cccc8f471798caac2c88)
- Update the build badge in the readme to point to this workflow.

#### Considerations

I haven't implemented the master build failure Slack notifications we had with Travis CI.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+name: tests
+on: push
```
Comment (johnsyweb) [inline .github/workflows/test.yml:3]: Should we test on PR, too?
```suggestion
on: [push, pull_request]
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+jobs:
+  test:
+    name: Test (Ruby ${{ matrix.ruby }}, ${{ matrix.os }})
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ ubuntu-20.04 ]
```
Comment (johnsyweb) [inline .github/workflows/test.yml:10]: Is it worth having this in the matrix with only one entry? Perhaps `runs-on: ubuntu-latest` may be less maintenance?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        ruby: [ '2.3', '2.4', '2.5', '2.6', '2.7' ]
```
Comment (johnsyweb) [inline .github/workflows/test.yml:11]: Is it worth testing against Ruby 3?

```suggestion
        ruby: [ '2.3', '2.4', '2.5', '2.6', '2.7', '3.0' ]
```

**Review by johnsyweb — APPROVED** [review]
I'm a fan. Thanks Orien.

File: .github/workflows/test.yml (hunk 3 above)
Comment (Domon) [inline .github/workflows/test.yml:11]: Does `event_sourcery` work with Ruby 3.0 released recently?

**Review by Domon — APPROVED** [review]
👍

File: .github/workflows/test.yml (hunk 1 above)
Comment (orien) [reply .github/workflows/test.yml:3]: 👍

File: .github/workflows/test.yml (hunk 2 above)
Comment (orien) [reply .github/workflows/test.yml:10]: Good idea. I'm all for less maintenance.

File: .github/workflows/test.yml (hunk 3 above)
Comment (orien) [reply .github/workflows/test.yml:11]: Yes, but I think we should do that in a dedicated PR. Perhaps work is required.

Comment (orien) [reply .github/workflows/test.yml:11]: I hope so. Let's explore in another PR.

File: .github/workflows/test.yml (hunk 2 above)
Comment (orien) [reply .github/workflows/test.yml]: Updated in e160298b8bae0828feab24079b79faadcb131b56

**Pull request merged**